auth/proofOfPossession.go

```diff
@@ -13,29 +13,22 @@ import (
 )
 
 var ErrNoClientCert = errors.New("there is no x509 client certificate provided")
-var ErrNoToken = errors.New("there is no token provided")
 
 // validateCertificate runs all proof of possession checks.
 // This ensures that the token was issued for the sender.
-func validateCertificate(clientCertificate *Certificate, token Token) error {
+func validateCertificate(clientCertificate *Certificate, token stdToken) error {
```

Comment:
in future this method should be public; with stdToken parameter this won't be possible, right?
Furthermore i don't understand that change from our authn.Token interface to stdToken struct?

Reply:
then we have to reimplement getCnfClaimMember in authn.Token as well.

Reply:
No, this won't be possible with stdToken. But there is no point in having an interface with private methods, since no other package will be able to satisfy this interface. That's why I removed the private methods and changed the parameters of internal functions to the actual struct. 
We might wanna think about dropping the interface alltogether, since it's an unnecessary abstraction.

Reply:
I have now removed the Interface and made the struct public, yet with private fields.

Reply:
mhh... but now - without the abstraction - don't we expose all methods provided with 3rd party jwt.Token, this is what you originally tried to avoid, isn't it?

Reply:
No, since the jwtToken field is still private other packages can't access it.